### PR TITLE
Hang on multi-line comments fixed.

### PR DESCRIPTION
**Pull request by seeraven**
Uncrustify hangs when indenting bad formatted multi-line comments like in
```.cpp
void function()
{
    printf( "Hello World\n" );
        /*
 //test
        /// Another comment
//end test
*/
}
```

This was caused by a subtraction without boundary check, so that the unsigned variable `ccol` actually got huge values. By ensuring the subtraction will not result in negative values this problem was fixed.

A test was added which shows the original problem.

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -1492,7 +1492,12 @@ static void output_comment_multi(chunk_t *pc)
              * This is not the first line, so we need to indent to the
              * correct column. Each line is indented 0 or more spaces.
              */
-            ccol -= col_diff;
+            // Ensure ccol is not negative
+            if (int(ccol) >= col_diff)
```
Comment (CDanU) [inline src/output.cpp]: A static _cast would be better here.

File: tests/config/comment-align-multiline.cfg [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+sp_func_def_paren=remove
+sp_inside_fparens=remove
+sp_inside_fparen=add
+sp_func_call_paren=remove
+sp_func_class_paren=remove
+sp_return_paren=add
+nl_start_of_file=remove
+nl_end_of_file=remove
+nl_struct_brace=add
+pos_arith=trail
```
Comment (CDanU) [inline tests/config/comment-align-multiline.cfg]: A config file is not needed for this testcase. Please use the `empty.cfg`

File: tests/cpp.test [hunk 3]
```diff
@@ -438,3 +438,5 @@
 34171  i1181.cfg                       cpp/i1181.cpp
 
 34190 bug_1003.cfg                     cpp/bug_1003.cpp
+
+34191 comment-align-multiline.cfg      cpp/comment-align-multiline.cpp
```
Comment (CDanU) [inline tests/cpp.test]:
```
34191 empty.cfg      cpp/comment-align-multiline.cpp
```

File: tests/output/cpp/34191-comment-align-multiline.cpp [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+void function()
+{
+    printf( "Hello World\n" );
+    /*
+       //test
+       /// Another comment
+       //end test
+     */
+}
```
Comment (CDanU) [inline tests/output/cpp/34191-comment-align-multiline.cpp]: Without the unnecessary changes (made by the config file) the output should be this:
```cpp
#include <stdio.h>

void function()
{
	printf( "Hello World\n" );
	/*
	   //test
	   /// Another comment
	   //end test
	 */
}

```
(Note that uncrustify defaults to tab indents)

**Comment by guy-maurel** [comment]
You can prevent the failure by using ./scripts/Run_uncrustify_for_sources.sh

**Comment by seeraven** [comment]
Good points! I've corrected the commit according to your suggestions.
